### PR TITLE
Init RSpec

### PR DESCRIPTION
**Pull request by falusi94**
Init [RSpec](https://github.com/rspec/rspec-rails) with [shoulda-matchers](https://github.com/thoughtbot/shoulda-matchers) and add a demonstrative example that fixes this [error](https://rollbar.com/salierri/pek-next/items/295/).

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,100 +1,27 @@
...
-# light-weight as possible. Requiring heavyweight dependencies from this file
-# will add to the boot time of your test suite on EVERY test run, even for an
-# individual file that may not need all of that loaded. Instead, consider making
-# a separate helper file that requires the additional dependencies and performs
-# the additional setup, and require it from the spec files that actually need
-# it.
-#
+# frozen_string_literal: true
+
+ENV['RAILS_ENV'] ||= 'test'
```
Comment (emanreyalp) [inline spec/spec_helper.rb]: Why is this necessary? 
It can happen when the file is loaded and the `RAILS_ENV` is not 'test'?

File: spec/requests/group_history_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    subject { get "/groups/#{group.id}/history" }
+
+    before { login_as(user) }
+
+    context 'when everything is fine' do
+      before { group.evaluations << create(:evaluation, point_request_status: Evaluation::ACCEPTED) }
+      before { subject }
+
+      it 'works fine' do
```
Comment (emanreyalp) [inline spec/requests/group_history_controller_spec.rb]: You can write it in one line if you want.
```suggestion
      it ('works fine') { expect(response).to have_http_status(:ok) } 
```
But maybe your way is more readable. :)

Comment (falusi94) [reply spec/requests/group_history_controller_spec.rb]: I wanted to use a shoulda matcher instead of this (and also for the other test), but idk yet why it does not work. I suspect it is only for controller tests, which would make sense, but it is working for this `it { should render_template :show }`.

I mainly added the draft tag because of this.

File: spec/spec_helper.rb (hunk 1 above)
Comment (falusi94) [reply spec/spec_helper.rb]: I think you can run easily with other environment.

I was thinking about adding some kind of db truncate (I am not sure yet how we should build the whole testing env) and it could be some safety stuff with `abort 'Running in prod env' if Rails.env.production?` to prevent production db truncation.

File: spec/requests/group_history_controller_spec.rb (hunk 2 above)
Comment (falusi94) [reply spec/requests/group_history_controller_spec.rb]: So now it is clear..
https://github.com/thoughtbot/shoulda-matchers#availability-of-rspec-matchers-in-example-groups

File: spec/support/authentication_helper.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module AuthenticationHelper
+  def login_as(user)
+    OmniAuth.config.test_mode = true
+    OmniAuth.config.add_mock(:oauth, auth_hash_for(user))
+    Rails.application.env_config['omniauth.auth'] = OmniAuth.config.mock_auth[:oauth]
+    get '/auth/oauth/callback'
```
Comment (emanreyalp) [inline spec/support/authentication_helper.rb]: It is just a little change :)
```suggestion
    get auth_oauth_callback_path
```
